routing/index_graph_starter.cpp

```diff
@@ -40,20 +40,34 @@ m2::PointD const & IndexGraphStarter::GetPoint(Joint::Id jointId)
   return m_graph.GetPoint(jointId);
 }
 
-m2::PointD const & IndexGraphStarter::GetPoint(RoadPoint const & rp)
+DirectedEdge const & IndexGraphStarter::FindFakeEdge(uint32_t fakeFeatureId)
 {
-  if (!IsFakeFeature(rp.GetFeatureId()))
-    return m_graph.GetPoint(rp);
-
   for (DirectedEdge const & e : m_fakeZeroLengthEdges)
   {
     ASSERT_EQUAL(GetPoint(e.GetFrom()), GetPoint(e.GetTo()), ());
-    if (e.GetFeatureId() == rp.GetFeatureId())
-      return GetPoint(e.GetFrom());
+    if (e.GetFeatureId() == fakeFeatureId)
+      return e;
   }
-  CHECK(false,
-        (rp, "is a point of a fake feature but it's not contained in |m_fakeZeroLengthEdges|"));
-  return m_graph.GetPoint(rp);
+  CHECK(false, ("Fake feature:", fakeFeatureId, "has to be contained in |m_fakeZeroLengthEdges|"));
+  return m_fakeZeroLengthEdges.front();
```

Comment:
Yes. But in this case there's a warning about returning local value by const ref if to do so.